### PR TITLE
Update stale_read doc for the support of TiFlash

### PR DESCRIPTION
**Pull request by hehechen**
Signed-off-by: hehechen <awd123456sss@gmail.com>

### What is changed, added or deleted? (Required)
Update stale_read doc since TiFlash supports stale read.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: stale-read.md [hunk 1]
```diff
@@ -24,3 +24,24 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
     - 指定时间范围：在会话级别中，如需 TiDB 在后续的查询中读取一个时间范围内尽可能新的数据并且不破坏隔离级别，你可以通过设置一个 session 变量 `tidb_read_staleness` 来指定一个时间范围。要使用该方式，请参阅[通过系统变量 `tidb_read_staleness` 读取历史数据](/tidb-read-staleness.md)。
 
 除此以外，你也可以通过设置系统变量 [`tidb_external_ts`](/system-variables.md#tidb_external_ts-从-v640-版本开始引入) 来在某一会话或全局范围读取某一时间点前的历史数据。要使用该方式，请参阅[通过系统变量 `tidb_external_ts` 读取历史数据](/tidb-external-ts.md)。
+
+## 限制
+当 Stale Read 查询下发到 TiFlash，且在设置的读取时间戳之后对该表做过 DDL 操作，此查询会报错。
```
Comment (flowbehappy) [inline stale-read.md:29]:
```suggestion
当 Stale Read 查询下发到 TiFlash，且在设置的读取时间戳之后对该表做过 DDL 操作，此查询会报错。原因是 TiFlash 只支持使用最新的表结构读取数据。
```

File: stale-read.md [hunk 2]
```diff
@@ -24,3 +24,24 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
     - 指定时间范围：在会话级别中，如需 TiDB 在后续的查询中读取一个时间范围内尽可能新的数据并且不破坏隔离级别，你可以通过设置一个 session 变量 `tidb_read_staleness` 来指定一个时间范围。要使用该方式，请参阅[通过系统变量 `tidb_read_staleness` 读取历史数据](/tidb-read-staleness.md)。
 
 除此以外，你也可以通过设置系统变量 [`tidb_external_ts`](/system-variables.md#tidb_external_ts-从-v640-版本开始引入) 来在某一会话或全局范围读取某一时间点前的历史数据。要使用该方式，请参阅[通过系统变量 `tidb_external_ts` 读取历史数据](/tidb-external-ts.md)。
+
+## 限制
```
Comment (qiancai) [inline stale-read.md:28]:
```suggestion
## 限制

```

File: stale-read.md [hunk 3]
```diff
@@ -24,3 +24,24 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
...
+当 Stale Read 查询下发到 TiFlash，且在设置的读取时间戳之后对该表做过 DDL 操作，此查询会报错。原因是 TiFlash 只支持使用最新的表结构读取数据。
```
Comment (qiancai) [inline stale-read.md:29]:
```suggestion
当对表的 Stale Read 查询下推到 TiFlash 时，如果该表在 Stale Read 所指定的读取时间戳之后执行过 DDL 操作，此查询将会报错。原因是 TiFlash 只支持从使用最新的表结构读取数据。

```

File: stale-read.md [hunk 4]
```diff
@@ -24,3 +24,24 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
...
+举个例子：
```
Comment (qiancai) [inline stale-read.md:30]:
```suggestion
例如：

```

File: stale-read.md (hunk 2 above)
Comment (qiancai) [reply stale-read.md:28]: Markdown 要求每个段落之间相隔一个空行

File: stale-read.md [hunk 5]
```diff
@@ -24,3 +24,24 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
...
+```sql
+create table t1(id int);
+alter table t1 set tiflash replica 1;
+```
+一分钟后进行 DDL 操作
```
Comment (qiancai) [inline stale-read.md:35]:
```suggestion

一分钟后进行 DDL 操作：

```

File: stale-read.md [hunk 6]
```diff
@@ -24,3 +24,24 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
...
+```sql
+alter table t1 add column c1 int not null;
+```
+然后 Stale Read 读取一分钟前的数据
```
Comment (qiancai) [inline stale-read.md:39]:
```suggestion

然后使用 Stale Read 读取一分钟前的数据：

```

File: stale-read.md [hunk 7]
```diff
@@ -24,3 +24,24 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
...
+```sql
+set @@session.tidb_enforce_mpp=1;
+select * from t1 as of timestamp NOW() - INTERVAL 1 minute;
+```
+此时 TiFlash 会报错
```
Comment (qiancai) [inline stale-read.md:44]:
```suggestion

此时 TiFlash 会报错：

```

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]

File: stale-read.md [hunk 8]
```diff
@@ -24,3 +24,35 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
     - 指定时间范围：在会话级别中，如需 TiDB 在后续的查询中读取一个时间范围内尽可能新的数据并且不破坏隔离级别，你可以通过设置一个 session 变量 `tidb_read_staleness` 来指定一个时间范围。要使用该方式，请参阅[通过系统变量 `tidb_read_staleness` 读取历史数据](/tidb-read-staleness.md)。
 
 除此以外，你也可以通过设置系统变量 [`tidb_external_ts`](/system-variables.md#tidb_external_ts-从-v640-版本开始引入) 来在某一会话或全局范围读取某一时间点前的历史数据。要使用该方式，请参阅[通过系统变量 `tidb_external_ts` 读取历史数据](/tidb-external-ts.md)。
+
+## 限制
+
+当对表的 Stale Read 查询下推到 TiFlash 时，如果该表在 Stale Read 所指定的读取时间戳之后执行过 DDL 操作，此查询将会报错。原因是 TiFlash 只支持从使用最新的表结构读取数据。
```
Comment (hfxsd) [inline stale-read.md:30]:
```suggestion
当对表的 Stale Read 查询下推到 TiFlash 时，如果该表在 Stale Read 所指定的读取时间戳之后执行过 DDL 操作，此查询将会报错。原因是 TiFlash 只支持从最新的表结构读取数据。
```

File: stale-read.md [hunk 9]
```diff
@@ -24,3 +24,35 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
...
+select * from t1 as of timestamp NOW() - INTERVAL 1 minute;
+```
+
+此时 TiFlash 会报错：
+
+```
+ERROR 1105 (HY000): other error for mpp stream: From MPP<query:<query_ts:1673950975508472943, local_query_id:18, server_id:111947, start_ts:438816196526080000>,task_id:1>: Code: 0, e.displayText() = DB::TiFlashException: Table 323 schema version 104 newer than query schema version 100, e.what() = DB::TiFlashException,
+```
+
+把 Stale Read 指定的读取时间戳改成 DDL 操作时间之后，就可以避免这个错误。
```
Comment (hfxsd) [inline stale-read.md:58]:
```suggestion
把 Stale Read 指定的读取时间戳改成 DDL 的操作时间，即可避免该错误。
```

File: stale-read.md [hunk 10]
```diff
@@ -24,3 +24,35 @@ TiDB 提供语句级别、会话级别以及全局级别的 Stale Read 使用方
...
+
+```sql
+set @@session.tidb_enforce_mpp=1;
+select * from t1 as of timestamp NOW() - INTERVAL 1 minute;
+```
+
+此时 TiFlash 会报错：
+
+```
+ERROR 1105 (HY000): other error for mpp stream: From MPP<query:<query_ts:1673950975508472943, local_query_id:18, server_id:111947, start_ts:438816196526080000>,task_id:1>: Code: 0, e.displayText() = DB::TiFlashException: Table 323 schema version 104 newer than query schema version 100, e.what() = DB::TiFlashException,
```
Comment (hfxsd) [inline stale-read.md:55]: 这里的 Table 323 是否需要和 L35 中的 t1 保持一致？

File: stale-read.md (hunk 9 above)
Comment (JaySon-Huang) [reply stale-read.md:58]: "把 Stale Read 指定的读取时间戳改成 DDL 操作完成之后时间，即可避免该错误。" 这样可能准确一些

**Review by hfxsd — APPROVED** [review]

**Pull request merged**